pkg/ccl/streamingccl/streamingest/stream_ingestion_job.go

```diff
@@ -270,13 +291,47 @@ func ingest(ctx context.Context, execCtx sql.JobExecContext, ingestionJob *jobs.
 	return errors.CombineErrors(ingestWithClient(), client.Close(ctx))
 }
 
+func ingestWithRetries(
+	ctx context.Context, execCtx sql.JobExecContext, ingestionJob *jobs.Job,
+) error {
+	ro := retry.Options{
+		InitialBackoff: 3 * time.Second,
+		Multiplier:     2,
+		MaxBackoff:     10 * time.Second,
+		MaxRetries:     5,
+	}
+
+	var err error
+	retryCount := 0
+	for r := retry.Start(ro); r.Next(); {
+		status := "start stream ingestion"
+		if retryCount != 0 {
+			status = fmt.Sprintf("retrying stream ingestion in the %d round with previous error: %s", retryCount, err)
+		}
+		updateRunningStatus(ctx, ingestionJob, status)
+		err = ingest(ctx, execCtx, ingestionJob)
```

Comment:
These statuses are going to be pretty rapidly overwritten by the status' we set in ingest().  Perhaps we should just set the status after we get the err so that the user knows that we've gotten an error and are waiting to retry.